Changed occurences of "data table" to "table" to reflect the componen…

ncameronbritt opened the pull request:
…t name.

Partially addresses https://github.com/patternfly/patternfly-next/issues/1993

LHinson commented:
@mcarrano want to take a look at this one as well? 

@ncameronbritt looks like you got conflicts :/ Do you use sourcetree to send PRs? If so, I can help you. If not, let me know what you use and we'll get someone else to help you!

ncameronbritt commented:
@LHinson I do use sourcetree, but it seems to be jacked up on my new machine.

LHinson commented:
thank you @jschuler!

gdoyle1 commented on packages/patternfly-4/content/design-guidelines/usage-and-behavior/lists-and-tables/design.md:
```diff
@@ -2,14 +2,14 @@
 * Content displayed may vary between rows
 
-Use a [data table](#data-tables) when:
+Use a [tables](#tables) when:
```
This should be "Use [tables](#tables) when:" -- remove the "a"

gdoyle1 commented:
@ncameronbritt Only found one thing to fix here! Also we say "data table" here https://www.patternfly.org/v4/design-guidelines/usage-and-behavior/toolbar#examples - could you include that fix in this PR as well?

ncameronbritt commented:
Addressed comments. The [html demo data table](https://www.patternfly.org/v4/documentation/core/demos/datatable) still needs to be done. I'm happy to do it, but I would need some guidance--I can't find the file in my for of the patternfly-org repo.

gdoyle1 commented:
@ncameronbritt Thanks! I'll check with the team to see how we can change that demo. I'll create a separate issue though and just reference this one!

jschuler commented on packages/patternfly-4/content/design-guidelines/usage-and-behavior/lists-and-tables/design.md:
```diff
@@ -2,14 +2,14 @@
 ---
 # Lists and tables
-PatternFly offers two components for displaying large data sets: the data list and the data table. While they satisfy similar use cases, choosing the correct component to use in your design will be dependent on the type of data you need to display.
+PatternFly offers two components for displaying large data sets: [data list](https://www.patternfly.org/v4/documentation/react/components/datalist/) and [table](https://www.patternfly.org/v4/documentation/react/components/table/). While they satisfy similar use cases, choosing the correct component to use in your design will be dependent on the type of data you need to display.
```
we should use relative URLs here

ncameronbritt replied:
Happy to make that change. Just tell me what to put for the url. :-)

jschuler commented:
@ncameronbritt @gdoyle1 I modified the URLs to be relative. Does the PR look good to you?

gdoyle1 commented:
@jschuler this looks good - data table looks like it still appears in core demos although it has been changed on their workspace. Is there a way we can fix this as well? Or should I create a separate issue?

jschuler commented:
@gdoyle1 yes a separate issue would be good. If not already done so, this should first be fixed upstream in @patternfly/patternfly-next repo, then when we can update the site to pull it in

gdoyle1 commented:
@jschuler Thanks! I believe it has been fixed in the patternfly-next repo already but will check. This is all set then

ncameronbritt commented:
lgtm. Thanks @jschuler
